ElectronClient/app/gui/NoteText.jsx

```diff
@@ -89,7 +89,7 @@ class NoteTextComponent extends React.Component {
 		this.localSearchDefaultState = {
 			query: '',
 			selectedIndex: 0,
-			resultCount: 0,
+			result: { query: '', count: 0 },
 		};
```

Comment:
What is the difference between `query` and `result.query`, and could the code be changed so that we don't need to have both? Also in general it's better to keep structures as flat as possible, as updating deeply nested ones is often more complicated (especially since we don't use a lib like Immutable).

Reply:
`query` is the query text in the search box.  `result.query` is the query text that corresponds to the current `result.count`.

`result.query` is needed because the search box is rendered twice upon each keypress. Upon the first render, the  `query` is up to date but the the search result `result.count` is not. With this mismatch, the search box can be rendered incorrectly. The second render is always correct, because it happens when the search result is updated.  Even though the render is incorrect only for a short while, it is noticeable.

If we can skip the first render, then the `result.query` wouldn't be needed. But I didn't see and easy way to do it.

Reply:
> Also in general it's better to keep structures as flat as possible,

Understood.  I will flatten the structure.